spec/lib/sendgrid_helper_spec.rb

```diff
@@ -2,62 +2,116 @@
 require 'sendgrid_helper'
 
 RSpec.describe SendgridHelper do
-  context "spam reports" do
-    context "error handling" do
-      [Curl::Err::CurlError, JSON::ParserError].each do |exception_class|
-        it "marks the email as valid if there's an error (#{exception_class})" do
-          expect(Curl::Easy).to receive(:perform).and_raise(exception_class)
+  context 'with sendgrid configured' do
+    around do |example|
+      smtp_settings = Rails.configuration.action_mailer.smtp_settings
+      Rails.configuration.action_mailer.smtp_settings = {
+        user_name: 'test_smtp_username',
+        password: 'test_smtp_password'
+      }
+      example.run
+      Rails.configuration.action_mailer.smtp_settings = smtp_settings
+    end
+
+    context "spam reports" do
+      context "error handling" do
+        [Curl::Err::CurlError, JSON::ParserError].each do |exception_class|
+          it "marks the email as valid if there's an error (#{exception_class})" do
+            allow(Curl::Easy).to receive(:perform).and_raise(exception_class)
+            expect(SendgridHelper.spam_reported?('test@example.com')).to be_falsey
```

Comment:
Regarding the second, to be honest, I'd rather rewrite this 'Helper' entirely to be (a) better tested and (b) not a module: I tried to make minimal changes here, and didn't rewrite the specs beyond some indentation and actually making them test what they claimed.

The first point is correct: it looks like they do always return a boolean. However, I suspect that having to use either `be true` or `be_truthy` is probably the fingerprint of a bad API, and it should be possible to do something like `expect(subject).not_to be_spam`.